controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -125,6 +127,27 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		// Error reading the object - requeue the request.
 		return ctrl.Result{}, errors.Wrap(err, "could not load host data")
 	}
+	// Fetch the HostData
+	hostData := &metal3v1alpha1.HostData{}
+	hostDataKey := client.ObjectKey{
+		Name:      host.Name,
+		Namespace: host.Namespace,
+	}
+	err = r.Get(ctx, hostDataKey, hostData)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		fmt.Println("Handle it")
+	}
```

Comment:
It's a tricky one. The failure could be due to e.g. missing CRD or RBAC permissions. Do we want to prevent all updates to BMHs in those circumstances? I suspect not, we probably just want to log the error and continue (but not attempt to create a replacement or anything).